proposals/bip-001.md

```diff
@@ -0,0 +1,107 @@
+# BIP Purpose and Guidelines
+
+| Field | Value |
+| --- | --- |
+| BIP | 1 |
+| Title | BIP Purpose and Guidelines |
+| Author | Alfredo Beaumont |
+| Status | Draft |
+| Created | 2017-05-26 |
+| Updated | 2017-05-26 |
+
+## Abstract
+
+BIP stands for Babelfish Improvement Proposal. A BIP is a design document which
+describes a change proposal to Babelfish. It's initially designed specifically
+to propose changes to [UAST](https://doc.bblf.sh/uast/specification.html)
+components regarding code categorization through Roles or related entities, but
+its scope may become broader in the future. It's informally defined.
+
+The BIP should provide a concise technical specification of, and rational for,
+the proposed change.
+
+We intend BIPs to be the primary mechanism for proposing new features or changes
+to existing ones in the UAST components.
+
+## Rationale
+
+Currently changes to UAST are mostly language driven, i. e., whenever a language
+driver implementor finds that there's some feature missing to complete an AST to
+UAST conversion, this new feature is requested to be added through a Pull
+Request. While this process is quick, it doesn't scale well: a single language
+need is not the best background for a universal feature request discussion, and
+it easies a process that may degenerate in a UAST that becomes the sum of each
+single language feature, which is something we should avoid. That's why
+introducing a more formal process, requiring a larger background and use cases,
+should help visualising which features are general enough, which ones are just
+special cases of a more general ones, which ones are just details that may be
+left for a future version, etc.
```

Comment:
I think that every feature shold be mapped to a general category, they should be as many categories as needed to achieve that, but not more. What we should not try to achieve is to have any single feature in every language as its own category, since that's unmanageable. That's only my opinion, it'd be nice to know if we all agreen on this, and then I can add some comments clearifying this point.